### PR TITLE
Resolves #2725

### LINKED ISSUE #2725: Add reference to `Set-AdfsFarmInformation -RemoveNode`
It'd be really useful to include a reference to **`Set-AdfsFarmInformation -RemoveNode`** here. While noting that `Remove-AdfsFarmNode` has been deprecated in favour of `Uninstall-WindowsFeature`, the latter cmdlet will remove the ADFS role from a server but it doesn't remove the node from the ADFS farm, leaving some residual configuration behind (on SQL-backed farms at least).

It took me a while to find any reference to using `Set-AdfsFarmInformation`, and through various discussion platforms I can see many others before me have run into the same issue.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 366c8c32-36b4-e43b-063a-610fa2dc2b9d
* Version Independent ID: 5b84c2d6-b201-39f0-bbbd-80fcfc57ad3b
* Content: [Remove-AdfsFarmNode (ADFS)](https://docs.microsoft.com/en-us/powershell/module/adfs/remove-adfsfarmnode?view=windowsserver2019-ps)
* Content Source: [docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/master/docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md)
* Product: **w10**
* Technology: **windows**
* GitHub Login: @JasonGerend
* Microsoft Alias: **jgerend**

### PR DESCRIPTION
**Pull request by velkovb**
Add note for proper node cleanup.
https://github.com/MicrosoftDocs/windows-powershell-docs/issues/2725

### CODE REVIEW

File: docset/winserver2012r2-ps/adfs/Remove-AdfsFarmNode.md [hunk 1]
```diff
@@ -27,8 +27,8 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
-Instead, use the Uninstall-WindowsFeaturehttp://go.microsoft.com/fwlink/?LinkID=287572 cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup please use the **Set-AdfsFarmInformation** with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/adfs/Remove-AdfsFarmNode.md:31]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2016-ps/adfs/Remove-AdfsFarmNode.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup please use the **Set-AdfsFarmInformation** with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup please use the **Set-AdfsFarmInformation** with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2022-ps/adfs/Remove-AdfsFarmNode.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup please use the **Set-AdfsFarmInformation** with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review.

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by get-itips — COMMENTED** [review]
when you say _However, this cmdlet..._ you refer to _Uninstall-WindowsFeature_, right? I wonder if that is 100% clear in the sentence

**Comment by velkovb** [comment]
I was a bit hesitant about using 'However' but couldn't think of anything better.

**Comment by get-itips** [comment]
> I was a bit hesitant about using 'However' but couldn't think of anything better.

@JohanFreelancer9 any ideas to improve the sentence? or leave as is?

**Review by tseward — APPROVED** [review]

File: docset/winserver2012r2-ps/adfs/Remove-AdfsFarmNode.md [hunk 5]
```diff
@@ -27,8 +27,8 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
-Instead, use the Uninstall-WindowsFeaturehttp://go.microsoft.com/fwlink/?LinkID=287572 cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/adfs/Remove-AdfsFarmNode.md:31]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. Note that the **Uninstall-WindowsFeature** cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2016-ps/adfs/Remove-AdfsFarmNode.md [hunk 6]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. Note that the **Uninstall-WindowsFeature** cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md [hunk 7]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. Note that the **Uninstall-WindowsFeature** cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

File: docset/winserver2022-ps/adfs/Remove-AdfsFarmNode.md [hunk 8]
```diff
@@ -29,7 +29,7 @@ Remove-AdfsFarmNode -GroupServiceAccountIdentifier <String> [-Credential <PSCred
 ## DESCRIPTION
 The **Remove-AdfsFarmNode** cmdlet is deprecated in this release.
 Instead, use the [Uninstall-WindowsFeature](https://go.microsoft.com/fwlink/?LinkID=287572) cmdlet.
-For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`.
+For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. However, this cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/adfs/Remove-AdfsFarmNode.md:32]:
```suggestion
For more information on the **Uninstall-WindowsFeature** cmdlet, type `Get-Help Uninstall-WindowsFeature`. Note that the **Uninstall-WindowsFeature** cmdlet removes only the server role but doesn't remove the node from the ADFS farm. For proper cleanup, use the **Set-AdfsFarmInformation** cmdlet with the `-RemoveNode` parameter.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
@velkovb, @get-itips, my suggestion to make the sentence more clear.

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by dariomws — APPROVED** [review]

**Comment by e0i** [comment]
Cleared assignees by mistake with this one. The PR is ready to be merged. Thanks. @tiburd

**Pull request merged**